### PR TITLE
Fixing some tests

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
- Fixes tests:
framework/DataObjects.dataObject_metadata
framework/DataObjects.load_csv
framework/ErrorChecks.badHistory
framework/ErrorChecks.badPointSet
framework/Samplers/Restart.CSV

- Still need work:
framework/ErrorChecks.unsyncedTypicalHistory
framework/PostProcessors/RavenOutputPostProcessor.dynamic
framework/PostProcessors/RavenOutputPostProcessor.invalidXMLtag
framework/PostProcessors/RavenOutputPostProcessor.static
framework/Optimizers.Infinite
framework.external_xml

Also fixes XSD validation so we can see qsub tests running.

### CODE REVIEW

File: framework/Steps.py [hunk 1]
```diff
@@ -552,6 +552,16 @@ def _localInitializeStep(self,inDictionary):
...
+    # check that no input data objects are also used as outputs?
+    for out in inDictionary['Output']:
+      if out.type not in ['PointSet','HistorySet','DataSet']:
+        continue
+      for inp in inDictionary['Input']:
+        if inp.type not in ['PointSet','HistorySet','DataSet']:
+          continue
+        if inp == out:
+          self.raiseAnError(IOError,'The same data object should not be used as both <Input> and <Output> in the same MultiRun step! ' \
+              + 'Step: "{}", DataObject: "{}"'.format(self.name,out.name))
```
Comment (wangcj05) [inline framework/Steps.py]: @alfoa @mandd Are you ok with this changes?

File: tests/framework/Samplers/Restart/gold/csv/CSVdump.csv [hunk 2]
```diff
@@ -1,26 +1,16 @@
...
-1.3879107403075115,2.186260211324845,1.0,1.0,7
-1.6697460405225275,2.3455607251603063,1.0,1.0,8
-1.9355390728301227,2.396767468749724,1.0,1.0,9
-1.8463109184164348,2.5388167320142534,1.0,1.0,10
-1.3132735130640849,2.4191945191517461,2.0,1.0,11
-1.5245481628283271,2.6852195008390627,2.0,1.0,12
-1.4434528938595794,2.2044522490362759,2.0,1.0,13
-1.2295772200984827,2.8781174372132208,2.0,1.0,14
-1.5344139073357019,2.0273875959746976,2.0,1.0,15
+x1,x2,ans
```
Comment (wangcj05) [inline tests/framework/Samplers/Restart/gold/csv/CSVdump.csv]: As we discussed, we should avoid regolding the tests, just do
```<what>input, output, metadata|ProbabilityWeight, .... </what>

File: tests/framework/Samplers/Restart/gold/mc/coarse.csv [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-1.9325573614175799,2.7203244894557459,1.0,1.0,2
-1.1281244478021106,2.00011438108052,1.0,1.0,3
-1.9990405156274886,2.3023325678199371,1.0,1.0,4
-1.2360889763189686,2.1467558925847419,1.0,1.0,5
-1.3965807262334462,2.0923385955608307,1.0,1.0,6
-1.3879107403075115,2.1862602113248455,1.0,1.0,7
-1.6697460405225275,2.3455607251603063,1.0,1.0,8
-1.9355390728301227,2.396767468749724,1.0,1.0,9
-1.8463109184164348,2.5388167320142538,1.0,1.0,10
+x1,x2,ans,ProbabilityWeight,ProbabilityWeight-x2,prefix,PointProbability,ProbabilityWeight-x1
```
Comment (wangcj05) [inline tests/framework/Samplers/Restart/gold/mc/coarse.csv]: same as before.

File: framework/Steps.py (hunk 1 above)
Comment (mandd) [reply framework/Steps.py]: is there a reason we need this?

Comment (PaulTalbot-INL) [reply framework/Steps.py]: Yes, we do need to do this.  

Even with restart sampling, this is the only way to ensure a dataobject has self-consistent sampling data.  I did discuss it with Andrea, and he agreed.  It only affected 4 tests in the repo, none of which actually needed to have the same DO as input and output in the same step.

File: tests/framework/Samplers/Restart/gold/csv/CSVdump.csv (hunk 2 above)
Comment (PaulTalbot-INL) [reply tests/framework/Samplers/Restart/gold/csv/CSVdump.csv]: The data I want to have as part of this test does not include the prefix or probability weights, so I regolded.

File: tests/framework/Samplers/Restart/gold/mc/coarse.csv (hunk 3 above)
Comment (PaulTalbot-INL) [reply tests/framework/Samplers/Restart/gold/mc/coarse.csv]: Yes, same as before.

**Comment by PaulTalbot-INL** [comment]
I added one more test to this, the Optimizer infinite case.  Since the UnorderedCsv can now handle inf and nan, we can do a better infinite test.

**Comment by wangcj05** [comment]
@PaulTalbot-INL Can you run your branch on Falcon, and report the test results here?

**Comment by PaulTalbot-INL** [comment]
Closing to add more test fixes to this PR.

**Pull request closed without merging**